devimint/src/gatewayd.rs

```diff
@@ -112,4 +121,71 @@ impl Gatewayd {
             .context("address must be a string")?
             .to_owned())
     }
+
+    pub async fn connect_to_peer(&self, pubkey: String, host: String) -> Result<()> {
+        cmd!(
+            self,
+            "lightning",
+            "connect-to-peer",
+            "--pubkey",
+            pubkey,
+            "--host",
+            host
+        )
+        .run()
+        .await?;
+        Ok(())
+    }
+
+    pub async fn get_funding_address(&self) -> Result<String> {
+        let address = cmd!(self, "lightning", "get-funding-address")
+            .out_string()
+            .await?;
+        Ok(address)
+    }
+
+    pub async fn open_channel(
```

Comment:
nit: are `connect_to_peer` and `open_channel` in devimint both necessary? Maybe `open_channel` could always call `connect_to_peer` first

Reply:
Good point! It looks like ldk-node's [`connect_open_channel`](https://docs.rs/ldk-node/0.2.1/ldk_node/struct.Node.html#method.connect_open_channel) method would support this and wouldn't run into issues if the peer is already connected (see [here](https://docs.rs/ldk-node/0.2.1/src/ldk_node/lib.rs.html#981)). However, it looks like this is not the case for LND and CLN. In order for this to work well, we'd need to be able to differentiate already-connected errors from internal or network errors. Thoughts?